ui/prometheus-plugin/src/components/PromQL.tsx

```diff
@@ -0,0 +1,45 @@
+// Copyright 2022 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import CodeMirror, { ReactCodeMirrorProps } from '@uiw/react-codemirror';
+import { PromQLExtension, CompleteConfiguration } from '@prometheus-io/codemirror-promql';
+import { EditorView } from '@codemirror/view';
+import { useTheme } from '@mui/material';
+import { useMemo } from 'react';
+
+export type PromQLEditorProps = { completeConfig: CompleteConfiguration } & Omit<
+  ReactCodeMirrorProps,
+  'theme' | 'extensions'
+>;
+
+export function PromQLEditor({ completeConfig, ...rest }: PromQLEditorProps) {
```

Comment:
There are some questions where this should live but I didn't feel this made sense in the `component` package. It feels like this is specific to the Prom plugin so it should live here. Open to opinions here though

Reply:
Yeah, I was definitely thinking this would end up here too, so no objections from me.